src/glados/db.py

```diff
@@ -0,0 +1,290 @@
+        """Insert an interaction object into the database.
+
+        Parameters
+        ----------
+        row : DataStoreInteraction
+            The row to be inserted
+        session : Session
+            session to be used
+
+        Returns
+        -------
+
+        """
+        session.add(interaction)
+        session.commit()
+        return interaction
+
+    def link_to_message_response(
+        self, interaction_id: str, message_response: dict, session: Session
+    ):
+        """Add info from the Slack message into the database for the interaction.
+
+        Parameters
+        ----------
+        interaction_id : str
+            The interaction ID that was returned on adding the message to the database.
+        message_response : dict
+            The raw message response from slack. The channel and ts will be pulled from this.
+        session: Session
+            session to be used
+
+        Returns
+        -------
+
+        """
+        ts_str = message_response.get("ts")
+        if not ts_str:
+            raise KeyError(f"ts missing from message body: {message_response}")
+        channel = message_response.get("channel", {}).get("id")
+        if not channel:
+            raise KeyError(f"channel missing from message body: {message_response}")
+        ts = datetime.fromtimestamp(ts_str)
+
+        self.update_interaction(
+            interaction_id, session, message_channel=channel, message_ts=ts
+        )
+
+    def link_to_message(
+        self, interaction_id: str, channel: str, ts: datetime, session: "Session"
+    ):
+        """Link to message by setting message ts and channel.
+
+        Parameters
+        ----------
+        interaction_id : str
+            interaction ID to link
+        channel : str
+            channel to link interaction to
+        ts : datetime
+            ts to link interaction to
+        session : Session
+            session to be used
+
+        Returns
+        -------
+
+        """
+        self.update_interaction(
+            interaction_id, session, message_channel=channel, message_ts=ts
+        )
+
+    def find_interaction_by_channel_ts(
+        self, channel: str, ts: datetime, session: Session
+    ) -> Optional[DataStoreInteraction]:
+        """Find the interaction in the datastore by channel and message ts.
+
+        Parameters
+        ----------
+        channel : str
+            channel of the interaction youre looking for
+        ts : datetime
+            ts of the interaction you are looking for
+        session : Session
+            session to be used
+
+        Returns
+        -------
+        DataStoreInteraction :
+            the interaction object
+
+        Raises
+        ------
+        ReferenceError :
+            There were more than one interaction that matched the channel and message_ts
+        """
+        query = session.query(DataStoreInteraction).filter(
```

Comment:
chain `.filter()` instead of using `and` directly in it. Also, you may look at doing `filter_by` instead of `filter` so you can just use the column attr instead:

`.filter_by(message_ts = ts).filter_by(message_channel = channel)`

https://docs.sqlalchemy.org/en/13/orm/query.html#sqlalchemy.orm.query.Query.filter_by